### PR TITLE
Add TestMode to Test Archetype with Live Default

### PR DESCRIPTION
**Pull request by alzimmermsft**

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml [hunk 1]
```diff
@@ -109,7 +110,10 @@ jobs:
         inputs:
           mavenPomFile: pom.xml
           ${{ insert }}: ${{ parameters.TestStepMavenInputs }}
-        env: ${{ parameters.EnvVars }}
+        env:
+          AZURE_TEST_MODE: ${{ parameters.EnvVars }}
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-tests.yml:114]:
```suggestion
          AZURE_TEST_MODE: ${{ parameters.TestMode }}
```

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml [hunk 2]
```diff
@@ -109,7 +110,10 @@ jobs:
...
+          ${{ each var in parameters.EnvVars }}:
```
Comment (benbp) [inline eng/pipelines/templates/jobs/archetype-sdk-tests.yml:115]: This won't handle AZURE_TEST_MODE being defined twice. I think you'd need to do it in a variable declaration like:

```
parameters:
  - name: TestMode
     type: string
     default: ""
variables:
  AZURE_TEST_MODE: $[ coalesce('${{ parameters.TestMode }}', 'LIVE') ]
```

Comment (alzimmermsft) [reply eng/pipelines/templates/jobs/archetype-sdk-tests.yml:115]: Should this change be made in the .NET configuration where I copied this from?

Comment (weshaggard) [reply eng/pipelines/templates/jobs/archetype-sdk-tests.yml:115]: @benbp I don't think your method would prevent that either.  I personally don't think we need to worry too much about this as if someone overrrides it via env variable it is a error and we will eventually fix it.

**Comment by alzimmermsft** [comment]
Validated that TestMode is passing AZURE_TEST_MODE into environment variables, but if children configurations pass `AZURE_TEST_MODE` in their EnvVars will will trigger a validation error

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml [hunk 3]
```diff
@@ -8,6 +8,7 @@ parameters:
   PreSteps: []
   PostSteps: []
   TimeoutInMinutes: 60
+  TestMode: ''
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-tests.yml:11]: I don't know if there are any other direct usages of this template but I think it is probably worth defaulting this to Live as well.

**Review by weshaggard — APPROVED** [review]
The change looks good and I think is a overall good move as it looks like a lot of live test pipelines were not actually running live tests.

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml (hunk 3 above)
Comment (alzimmermsft) [reply eng/pipelines/templates/jobs/archetype-sdk-tests.yml:11]: I'll circle back to this in a follow-up PR as not to trigger another ~300 pipelines

**Pull request merged**